src/readabilityScore/AbstractIndexCalculator.java

```diff
@@ -0,0 +1,36 @@
+package readabilityScore;
+
+public abstract class AbstractIndexCalculator implements IndexCalculator {
+    protected Text text;
+    protected Double index;
+    protected Integer age;
+
+    @Override
+    public void setText(Text text) {
+        this.text = text;
+        index = null;
+        age = null;
+    }
+
+    @Override
+    public Integer determineAge() {
```

Comment:
Looks like it's not a responsibility of `IndexCalculator` to determine the age based on index (the the related comment below). We can have a dedicated entity for that, e.g. `IndexToAgeConverter`